src/field/slot_machine.c

```diff
@@ -1339,6 +1339,69 @@ bool8 sub_8102EA4(struct Task *task)
     return FALSE;
 }
 
+extern bool8 (*const gUnknown_083ECB40[])(void);
+extern void (*const gUnknown_083ECB4C[])(void);
+
+bool8 sub_8102EC0(struct Task *task)
+{
+    task->data[0]++;
+    eSlotMachine->unk34[task->data[15]] = 0;
+    eSlotMachine->unk2E[task->data[15]] = 0;
+    if (eSlotMachine->unk0A == 0 && (eSlotMachine->unk04 == 0 || eSlotMachine->unk06 == 0 || !gUnknown_083ECB40[task->data[15]]()))
+    {
+        eSlotMachine->unk06 = 0;
+        gUnknown_083ECB4C[task->data[15]]();
+    }
+    task->data[1] = eSlotMachine->unk2E[task->data[15]];
+    return TRUE;
+}
+
+extern const u16 gUnknown_083ECB58[5]; // don't move this
```

Comment:
does the ordering matter?

Reply:
I plan to nuke the memcpy call later by replacing it with the commented line below